src/frontend/src/common/components/library/data_table/index.tsx

```diff
@@ -137,11 +148,42 @@ export const DataTable: FunctionComponent<Props> = ({
     });
   };
 
+  const headerCheckbox = (): React.ReactNode => {
+    function handleClick() {
+      handleHeaderCheckboxClick();
+    }
+    return (
+      <HeaderCheckbox
+        color="primary"
+        checked={isHeaderChecked}
+        onClick={handleClick}
+        indeterminate={isHeaderIndeterminant}
+      />
+    );
+  };
+
+  const rowCheckbox = (item: TableItem): React.ReactNode => {
+    let handleClick;
+    if (item !== undefined) {
+      const checked: boolean = checkedSamples.includes(item.publicId);
+      handleClick = function handleClick() {
+        handleRowCheckboxClick(
+          item.publicId.toString(),
```

Comment:
prefer `String(item.publicId)` in case `publicId` is `undefined`

Reply:
oh interesting i didn't know to use String() when dealing with undefined

Reply:
should i replace all instances of .toString with String()?

Reply:
Yeah that'd be great! Default to using `String()` and only use `.toString()` when absolutely necessary, such as when invoking a custom Class instance method that overrides `.toString()` to transform the class instance to a string.

E.g.,

```ts
Class CustomStuff {
  toString() {
    const saidString = // custom transform, such as serializing certain properties and ignoring private properties here
   return saidString;
  }
}
```

when we call `undefined.toString()` we get a runtime error, since `undefined` doesn't have a method called `toString()`!

Reply:
sounds good! Thanks for the explanation!!